src/client.c

```diff
@@ -59,34 +65,75 @@ int run_list_online_players(char *host, int port) {
 	return EXIT_SUCCESS;
 }
 
+void *player_lobby(void *_net_client) {
+	NetClient *net_client = (NetClient *)_net_client;
+	// get the list of possible opponents
+	StringArray *online_usernames = tetris_list(net_client);
+	while (online_usernames->length == 0) {
+		sleep(2);
+		online_usernames = tetris_list(net_client);
+	};
+
+	// null terminate the string array
+	string_array_resize(online_usernames, online_usernames->length + 1);
+
+	WidgetSelection *selection = ttviz_select(
+	    online_usernames->strings, online_usernames->length, "Opponent", 0);
+	StringArray *opponents = selection_to_string_array(selection);
+	tetris_opponent(net_client, opponents);
+
+	tetris_tell_server_to_start(net_client);
+
+	// free up resources
+	string_array_destroy(online_usernames);
+	selection_destroy(selection);
+
+	return NULL;
+}
+
 /**
  * run an online game of tetris
  */
-int run_online(char *host, int port, char *username, char *opponent) {
+int run_online(char *host, int port) {
```

Comment:
NIT: A TUI prompt for an address and port if `-a` is not specified should be added.

I'll create an issue for it if you agree.

Reply:
+1 Go ahead and create the issue; that makes sense to me! :smiley: 

In the future, I also envision that we might have a default server like ttetris.example.org if we get around to packaging, etc.

Reply:
See #33 

I do agree with this. Additionally, it may be cool to have a public server. However, we will need to take into account security risks before deploying such a thing.